### PR TITLE
Refactor to have scraper manager and add comidoc as source

### LINKED ISSUE #128: Add new source ?
Now that we have a good base, I'd like to explore the possibility of adding new source. What do you think @cullzie ?

### PR DESCRIPTION
**Pull request by cullzie**
# Description

**Setting as WIP for now as there are a lot of changes and I don't have a lot of time to test this fully ATM
Please feel free to leave a review and any comments 
Thanks :rocket:**

Refactor to use asyncio which improves speed of scrapers
Moved from `requests` to `aiohttp`
Create scraper manager which handles scrapers and their states
Added basescraper which any new scrapers should inherit from
Added comidoc source. Gets free courses in the past 5 days. (Can possibly make this configurable in future)
Added new cli arg `--debug`. Moved some of the info logging to be debug so console output is cleaner. If a user wants to see all messages they can call with this flag.
Added new cli arg `--comidoc`.  If this arg is passed we run comidoc scraper only
Added new cli arg `--tutorialbar`. If this arg is passed we run tutorialbar scraper only
^^ If neither of the args above is passed we run all scrapers 

Fixes #128 
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Run locally and unittests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] I have added tests that prove my fix is effective or that my feature works (Optional)
- [ ] New and existing unit tests pass locally with my changes (Optional)

### CODE REVIEW

**Comment by fakeid30** [comment]
I tested it . Couple suggestions @cullzie : 

1. Need to implement 1-2 sec delay after initial username and password input. Currently udemy shows an error sometimes when you click the login button too fast. (I suspect they have implemented it recently)
2. Need to update the documentation to have a toggle for one source or all . 
3. Need to update the CI to include the argument change.

**Comment by cullzie** [comment]
@fakeid30 ye I noticed that on login as well a while ago. I left a TODO in the code but I haven't got back around to it yet. 

I probably need to have a look at the cache hit logic as well. Since comidoc returns around a hundred links we don't want to stop the script if 12 are in the cache. Maybe we should remove it entirely and set the tutorialbar max pages default to 5. If someone wants to run all they can pass a higher number. What do you think?

**Comment by fakeid30** [comment]
How about 12 in tutorialbar and 50 in comidoc for cache ?

**Comment by cullzie** [comment]
For comidoc it is only one call so stopping after 50 cache hits offers no benefit. If they are all in the cache the check will complete pretty immediately even if checking 100, 200 etc links.

For tutorialbar the max-pages offers a similar outcome as cache-hits. If the default for that is 5 and since we have updated to asyncio it is very quick to fetch and check the cache for those entries. 
I think it's also clearer to an end user what is happening with max-pages over the cache-hits logic.
I'll do some comparisons but I don't think there will be a huge difference in speed

**Comment by fakeid30** [comment]
Ok sounds good to me

File: pyproject.toml [hunk 1]
```diff
@@ -1,22 +1,23 @@
 [tool.poetry]
 name = "automatic-udemy-course-enroller-get-paid-udemy-courses-for-free"
-version = "0.3"
+version = "1.0"
```
Comment (fakeid30) [inline pyproject.toml:3]: think we need version 2.0 for this patch

Comment (cullzie) [reply pyproject.toml:3]: I was just bumping here to have the current release version. When we go to release again we just need to remember to bump it again

Comment (fakeid30) [reply pyproject.toml:3]: Alright . Sounds good to me.

**Comment by cullzie** [comment]
> 1. Need to implement 1-2 sec delay after initial username and password input. Currently udemy shows an error sometimes when you click the login button too fast. (I suspect they have implemented it recently)

I've opened a separate task for this as I don't want to add too many changes here: #162

**Pull request merged**